devices/custom_devices_diy.js

```diff
@@ -143,6 +143,24 @@ const tzLocal = {
 };
 
 const fzLocal = {
+    // SNZB-02 reports stranges values sometimes
+    // https://github.com/Koenkk/zigbee2mqtt/issues/13640
+    SNZB02_temperature: {
+        ...fz.temperature,
+        convert: (model, msg, publish, options, meta) => {
+            if (msg.data.measuredValue > -3300 && msg.data.measuredValue < 10000) {
+                return fz.temperature.convert(model, msg, publish, options, meta);
+            }
+        },
+    },
+    SNZB02_humidity: {
```

Comment:
Can you move these two conveters to `fromZigbee.js` and remove `sonoff.js/SNZB02_temperature` converter (to prevent code duplication)

Reply:
It uses, slightly, different filter values, is this a problem?

Reply:
No, both are really extreme (-33 vs -100)